src/julia-syntax.scm

```diff
@@ -1209,13 +1217,15 @@
                   (argl       (car argl-stmts))
                   (name       (check-dotop (car argl)))
                   (argname    (if (overlay? name) (caddr name) name))
+                  ;;
                   ;; fill in first (closure) argument
+                  (self-name  (if (nodot-sym-ref? argname) argname (gensy)))
```

Comment:
I don't think this is right; `argname` might not be a symbol here, it can be a `::` expression. Using #self# is also ok because we want to avoid making too many gensyms.